pkg/module/sync/runner_test.go

```diff
@@ -232,59 +226,42 @@ func TestNeedToUpdate(t *testing.T) {
 			true,
 		},
 		{
-			"When channel switch, expect need to update",
+			"When new module version available and module is mandatory module, expect need to update",
 			args{
-				&v1beta2.Manifest{},
-				&v1beta2.Manifest{
-					ObjectMeta: apimetav1.ObjectMeta{
-						Labels: map[string]string{shared.ChannelLabel: "fast"},
-					},
-					Spec: v1beta2.ManifestSpec{Version: "0.1"},
-				}, &v1beta2.ModuleStatus{
-					Version: "0.1", Channel: "regular", Template: &v1beta2.TrackingObject{
-						PartialMeta: v1beta2.PartialMeta{
-							Generation: trackedModuleTemplateGeneration,
-						},
-					},
-				},
+				builder.NewManifestBuilder().WithName("test").WithVersion("0.1").WithChannel(
+					v1beta2.DefaultChannel).IsMandatoryModule().Build(),
+				builder.NewManifestBuilder().WithName("test").WithVersion("0.2").WithChannel(
+					v1beta2.DefaultChannel).IsMandatoryModule().Build(),
+				nil,
 				trackedModuleTemplateGeneration,
 			},
 			true,
 		},
 		{
-			"When cluster Manifest in divergent state, expect need to update",
```

Comment:
To Reviewer:

I also deleted this test case, based on https://github.com/kyma-project/lifecycle-manager/pull/1790/files#r1725307658